### PR TITLE
Separate linux kernel and glibc APIs as much as possible

### PR DESCRIPTION
**Pull request by joakim-noah**
This pull uses the new `CRuntime_Glibc` and `CRuntime_Bionic` predefined global versions from [dmd pulls #4111](https://github.com/D-Programming-Language/dmd/pull/4111) and [#3643](https://github.com/D-Programming-Language/dmd/pull/3643) to start separating out linux kernel APIs from glibc and bionic APIs.  I used dstep and a modified Dscanner to help find type and struct declarations, but had to check enums again by hand.  dmd pull #4111 will have to be merged before this one.

### CODE REVIEW

**Comment by joakim-noah** [comment]
Updated this pull so that the remaining linux/Android blocks use the appropriate `CRuntime` instead.  When combined with #1069, there are no more uses of `version(Android)` in druntime and all tests pass on both linux/Glibc/x86 and linux/Bionic/x86, ie Android/x86.  Some blocks had a mix of declarations from Glibc and the linux kernel, so I added a comment indicating that.  I also modified `core.sys.posix.sys.utsname` so that the initial `extern(C)` attribute is a label instead.

**Comment by joakim-noah** [comment]
Any reason this pull has been sitting unmerged for almost two months now?

**Comment by MartinNowak** [comment]
Lack of time, scheduled for review next week.

**Comment by joakim-noah** [comment]
No need to be sorry.  I was just worried that the controversy above was mistakenly presumed to be about this pull, when most of the discussion above was about changes made in #1069.

**Comment by joakim-noah** [comment]
Updated this pull to deal with the new modules `core.sys.linux.stdio` and `core.sys.posix.sys.msg`, plus amended a comment in `core.sys.windows.stat` to not refer to `std.c` from phobos anymore.  Not sure why `core.sys.posix.sys.msg` is in `core.sys.posix` when it only has declarations for linux.

**Comment by joakim-noah** [comment]
Ping, this pull was marked as ready for review more than two months ago when it passed the auto-tester.  Unless someone is planning on checking all the APIs to see where they actually come from, which I doubt since it took me days to do so in the first place, should be pretty straightforward to check for other, smaller mistakes and merge.

**Comment by joakim-noah** [comment]
Been a couple months, @MartinNowak, what is the status of this?  I know you've been busy pushing out the stable release, but hopefully this can get in now.

**Comment by joakim-noah** [comment]
Can someone merge this?  I don't see what the holdup is, most of it is simple renaming.

I'm about to submit another PR for Android/ARM support, but I need the `core.stdc.errno` portion of this pull to get merged first, so I can add ARM support to that new portion.

**Comment by schveiguy** [comment]
I noticed you removed all the docs for the errnos for linux. Was that on purpose?

**Comment by joakim-noah** [comment]
[Yep](https://github.com/D-Programming-Language/druntime/pull/1010#discussion_r22292013).

**Comment by schveiguy** [comment]
Ugh... How annoying. But I guess that's life.

File: src/core/sys/posix/time.d [hunk 1]
```diff
@@ -123,10 +123,11 @@ else version (Solaris)
 {
     enum CLOCK_MONOTONIC = 4;
 }
-else version (Android)
+else version (CRuntime_Bionic)
 {
-    enum CLOCK_MONOTONIC    = 1;
-    enum CLOCK_MONOTONIC_HR = 5;
+    enum CLOCK_MONOTONIC        = 1;
+    enum CLOCK_MONOTONIC_RAW    = 4;
```
Comment (MartinNowak) [inline src/core/sys/posix/time.d]: Better move this non-standard thing to core.sys.linux.time.CLOCK_MONOTONIC_RAW.

**Comment by MartinNowak** [comment]
> No need to be sorry. I was just worried that the controversy above was mistakenly presumed to be about this pull, when most of the discussion above was about changes made in #1069.

No, you finally seem to hit the sweet spot ;).

**Comment by ibuclaw** [comment]
This PR caused a regression. https://issues.dlang.org/show_bug.cgi?id=22689

**Pull request merged**